cpp/src/Ice/Properties.cpp

```diff
@@ -21,14 +21,12 @@ using namespace IceInternal;
 
 namespace
 {
-    /// Find a property in the Ice property set.
+    /// Find the Ice property array for a given property name.
     /// @param key The property name.
-    /// @param logWarnings Whether to log relevant warnings.
-    /// @return The property if found, nullopt otherwise.
-    optional<Property> findIceProperty(string_view key, bool logWarnings)
+    /// @return The property array if found, nullopt otherwise.
+    optional<const PropertyArray*> findIcePropertyArray(string_view key)
```

Comment:
Seems odd to use optional for a pointer, can't we just use `nullptr` for not set?